exploration/0006-number-selection.md

```diff
@@ -0,0 +1,127 @@
+such as a four-digit year or the integer value of an enumerator.
+
+Furthermore, as pointed out by <a href="https://github.com/ryzokuken">@ryzokuken</a>
+in <a href="https://github.com/unicode-org/message-format-wg/pull/457#discussion_r1307443288">#457 (comment)</a>,
+ordinal values are not truly "plural",
+even if they use categories with similar names and are determined by similar rules.
+
+## Use-Cases
+
+Put together, we have thus identified three different types of selection
+we would like to support on numeric values:
+
+- cardinal plural selection
+- ordinal selection
+- exact match selection
+
+## Requirements
+
+- Enable cardinal plural selection.
+- Enable ordinal number selection.
+- Enable exact match selection.
+- Support relevant formatting options, such as `minimumFractionDigits`.
+- Encourage developers to use the same options for formatting and selection.
+
+## Constraints
+
+ICU MF1 messages using `plural` and `selectordinal` should be representable in MF2.
+
+## Proposed Design
+
+Given that we already have a `:number`,
+it makes sense to add a `<matchSignature>` to it with an option
+
+```xml
+<option name="type" values="plural ordinal exact" default="plural" />
+```
+
+The default `plural` value is presuming that to be the most common use case,
+and it affords the least bad fallback when used incorrectly:
+Using "plural" for "exact" still selects exactly matching cases,
+whereas using "exact" for "plural" will not select LDML category matches.
+This might not be noticeable in the source language,
+but may cause problems in target languages that the original developer is not considering.
+
+Additional options such as `minimumFractionDigits` and others already supported by `:number`
+should also be supported.
+
+## Alternatives Considered
+
+An alternative approach to this problem could be to leave the `:number` `<matchSignature>` undefined,
+and to define three further functions, each with a `<matchSignature>`:
+
+- `:plural`
+- `:ordinal`
+- `:exact`
+
+which would each need the same set of options as `:number`, except for `type`.
+
+This approach would also mostly work, but it introduces new failure modes:
+
+- If a `:number` is used directly as a selector, this should produce a runtime error.
+- If a `:plural`/`:ordinal`/`:exact` is used as a formatter, this should produce a runtime error.
+- Developers are less encouraged to use the same formatting and selection options.
+
+To expand on the last of these,
+consider this message:
+
+```
+match {$count :plural minimumFractionDigits=1}
+when 0 {You have no apples}
+when 1 {You have exactly one apple}
+when * {You have {$count :number minimumFractionDigits=1} apples}
+```
+
+Here, because selection on `:number` is not allowed,
+it's easy to duplicate the options because _some_ annotation is required on the selector.
+It would also be relatively easy to leave out the `minimumFractionDigits=1` option from the selector,
+as it's not required for the English source.
```

Comment:
It does and that is good from the point of view that "one apple" is incorrectly internationalized with keyword `one`. You'll notice that elsewhere I did both 😉 

However, we should be careful to model good internationalization in our docs because others will copy us. In my example above I put the formatted count in the "one" example for that reason.